improver_tests/nowcasting/optical_flow/test_OpticalFlow.py

```diff
@@ -541,80 +540,6 @@ def test_values(self):
-        greater_than_10_percent_zeroes_array = np.array(
-            [[3.0, 5.0, 7.0], [0.0, 2.0, 1.0], [1.0, 1.0, 1.0]]
-        )
-        warning_msg = "cells within the domain have zero advection"
-        self.plugin._zero_advection_velocities_warning(
-            greater_than_10_percent_zeroes_array, self.rain_mask
-        )
-        self.assertTrue(any(item.category == UserWarning for item in warning_list))
-        self.assertTrue(any(warning_msg in str(item) for item in warning_list))
-
-    @ManageWarnings(record=True)
-    def test_no_warning_raised_if_no_zeroes(self, warning_list=None):
-        """Test that no warning is raised if the number of zero values in the
-        array is below the threshold used to define an excessive number of
-        zero values."""
-        nonzero_array = np.array([[3.0, 5.0, 7.0], [2.0, 2.0, 1.0], [1.0, 1.0, 1.0]])
-        self.plugin._zero_advection_velocities_warning(nonzero_array, self.rain_mask)
-        self.assertTrue(len(warning_list) == 0)
-
-    @ManageWarnings(record=True)
-    def test_no_warning_raised_if_fewer_zeroes_than_threshold(self, warning_list=None):
-        """Test that no warning is raised if the number of zero values in the
-        array is below the threshold used to define an excessive number of
-        zero values when at least one zero exists within the array."""
-        rain = np.ones((5, 5))
-        less_than_10_percent_zeroes_array = np.array(
-            [
-                [1.0, 3.0, 5.0, 7.0, 1.0],
-                [0.0, 2.0, 1.0, 1.0, 1.0],
-                [1.0, 1.0, 1.0, 1.0, 1.0],
-                [1.0, 1.0, 1.0, 1.0, 1.0],
-                [1.0, 1.0, 1.0, 1.0, 1.0],
-            ]
-        )
-        self.plugin._zero_advection_velocities_warning(
-            less_than_10_percent_zeroes_array, np.where(rain > 0)
-        )
-        self.assertTrue(len(warning_list) == 0)
-
-    @ManageWarnings(record=True)
-    def test_no_warning_raised_for_modified_threshold(self, warning_list=None):
-        """Test that no warning is raised if the number of zero values in the
-        array is below the threshold used to define an excessive number of
-        zero values when the threshold is modified."""
-        less_than_30_percent_zeroes_array = np.array(
-            [[3.0, 5.0, 7.0], [0.0, 2.0, 1.0], [0.0, 1.0, 1.0]]
-        )
-        self.plugin._zero_advection_velocities_warning(
-            less_than_30_percent_zeroes_array, self.rain_mask, zero_vel_threshold=0.3
-        )
-        self.assertTrue(len(warning_list) == 0)
-
-    @ManageWarnings(record=True)
-    def test_no_warning_raised_outside_rain(self, warning_list=None):
-        """Test warning ignores zeros outside the rain area mask"""
-        rain = np.array([[0, 0, 1], [0, 1, 1], [1, 1, 1]])
-        wind = np.array([[0, 0, 1], [0, 1, 1], [1, 1, 1]])
-        self.plugin._zero_advection_velocities_warning(wind, np.where(rain > 0))
-        self.assertTrue(len(warning_list) == 0)
```

Comment:
Added back in using following syntax:
```
        with warnings.catch_warnings():
            warnings.simplefilter(action="error", category=UserWarning)
```
Such that an error is raised if a UserWarning is raised and the test fails.